### ISSUE TITLE
Projecting VideoTexture

### ISSUE THREAD

**Issue by andrasf** [marcofugaro/three-projected-material#18]
Hi!

I would like to project a VideoTexture to a 3d model. Should it be possible with this library? In theory the classes act the same. But seems like only the right or bottom pixels are getting repeated in stripes.

Thanks,
András

**Comment by marcofugaro** [comment]
> the right or bottom pixels are getting repeated in stripes.

Hmm, this never happened to me before, could you share a reproducible example?

**Comment by andrasf** [comment]
Hi!

I modified 3d-model.html to show the problem. Maybe I am missing something very basic.

When I put the video into a canvas, and set the canvas 1px larger than the video, it looks like if the bottom 1 px of the video is a solid red color, the whole projected image is red.

Commit is here: https://github.com/marcofugaro/three-projected-material/compare/master...andrasf:master

András

**Comment by JamesBotterill** [comment]
I have a similar issue where the y axis isn't seemingly being scaled so you end up with what looks like a stripe, the video is playing as the colours change correctly.

**Comment by CptCompiler** [comment]
Hi @marcofugaro,
thank you very much for this very cool lib. I really appreciate your work.
Do you have any clue, why this could happen? 
Did you ever manage to get it running with a video texture?

Thanks

**Comment by CptCompiler** [comment]
The problem is in computeScaledDimensions.
`ratio` will be nan because texture.image is a video element that doesn't support `natrualWidth`
```
function computeScaledDimensions(texture, camera, textureScale, cover) {
  // return some default values if the image hasn't loaded yet
  if (!texture.image) {
    return [1, 1];
  }

  const ratio = texture.image.naturalWidth / texture.image.naturalHeight;
```

**Comment by marcofugaro** [comment]
Fixed in `v2.0.2`!

Thanks for the tip @CptCompiler, we now have a working video example:

https://marcofugaro.github.io/three-projected-material/video

**Comment by CptCompiler** [comment]
Hey @marcofugaro, thank YOU for your quick fix!!
I am so happy :-)
